hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/PrecompileFunctionSelector.java

```diff
@@ -0,0 +1,164 @@
+    // approve(address to, uint256 tokenId)
+    ABI_ID_ERC_APPROVE(0x95ea7b3),
+    // setApprovalForAll(address operator, bool approved)
+    ABI_ID_ERC_SET_APPROVAL_FOR_ALL(0xa22cb465),
+    // getApproved(uint256 tokenId)
+    ABI_ID_ERC_GET_APPROVED(0x081812fc),
+    // isApprovedForAll(address owner, address operator)
+    ABI_ID_ERC_IS_APPROVED_FOR_ALL(0xe985e9c5),
+    // ownerOf(uint256 tokenId)
+    ABI_ID_ERC_OWNER_OF_NFT(0x6352211e),
+    // tokenURI(uint256 tokenId)
+    ABI_ID_ERC_TOKEN_URI_NFT(0xc87b56dd),
+    // wipeTokenAccount(address, address, uint32)
+    ABI_WIPE_TOKEN_ACCOUNT_FUNGIBLE(0x9790686d),
+    // wipeTokenAccountNFT(address, address, int64[])
+    ABI_WIPE_TOKEN_ACCOUNT_NFT(0xf7f38e26),
+    // isFrozen(address token, address account)
+    ABI_ID_IS_FROZEN(0x46de0fb1),
+    // freezeToken(address token, address account)
+    ABI_ID_FREEZE(0x5b8f8584),
+    // unfreezeToken(address token, address account)
+    ABI_ID_UNFREEZE(0x52f91387),
+
+    // **** HIP-358 function selectors ****
+    // createFungibleToken(HederaToken memory token, uint initialTotalSupply, uint decimals)
+    ABI_ID_CREATE_FUNGIBLE_TOKEN(0x7812a04b),
+    // createFungibleTokenWithCustomFees(
+    //  HederaToken memory token,
+    //  uint initialTotalSupply,
+    //  uint decimals,
+    //  FixedFee[] memory fixedFees,
+    //  FractionalFee[] memory fractionalFees)
+    ABI_ID_CREATE_FUNGIBLE_TOKEN_WITH_FEES(0x4c381ae7),
+    // createNonFungibleToken(HederaToken memory token)
+    ABI_ID_CREATE_NON_FUNGIBLE_TOKEN(0x9dc711e0),
+    // createNonFungibleTokenWithCustomFees(
+    //  HederaToken memory token,
+    //  FixedFee[] memory fixedFees,
+    //  RoyaltyFee[] memory royaltyFees)
+    ABI_ID_CREATE_NON_FUNGIBLE_TOKEN_WITH_FEES(0x5bc7c0e6),
+
+    // **** HIP-514 function selectors ****
+    // getFungibleTokenInfo(address token)
+    ABI_ID_GET_FUNGIBLE_TOKEN_INFO(0x3f28a19b),
+    // getTokenInfo(address token)
+    ABI_ID_GET_TOKEN_INFO(0x1f69565f),
+    // getNonFungibleTokenInfo(address token, int64 serialNumber)
+    ABI_ID_GET_NON_FUNGIBLE_TOKEN_INFO(0x287e1da8),
+    // getTokenDefaultFreezeStatus(address token)
+    ABI_ID_GET_TOKEN_DEFAULT_FREEZE_STATUS(0xa7daa18d),
+    // getTokenDefaultKycStatus(address token)
+    ABI_ID_GET_TOKEN_DEFAULT_KYC_STATUS(0x335e04c1),
+    // isKyc(address token, address account)
+    ABI_ID_IS_KYC(0xf2c31ff4),
+    // 	grantTokenKyc(address  token, address account)
+    ABI_ID_GRANT_TOKEN_KYC(0x8f8d7f99),
+    // revokeTokenKyc(address token, address account)
+    ABI_ID_REVOKE_TOKEN_KYC(0xaf99c633),
+    // getTokenCustomFees(address token)
+    ABI_ID_GET_TOKEN_CUSTOM_FEES(0xae7611a0);
+
+    private final int functionSelector;
+
+    PrecompileFunctionSelector(int functionSelector) {
+        this.functionSelector = functionSelector;
+    }
+
+    public static PrecompileFunctionSelector fromFunctionId(int n) {
+        for (PrecompileFunctionSelector c : values()) {
```

Comment:
This should be O(log(n)) with a hashed lookup (like a `Map<Integer, PrecompileFunctionSelector>`) or a binary search on a sorted array, rather than a linear search (O(n)).

Reply:
PrecompileFunctionSelector was reverted